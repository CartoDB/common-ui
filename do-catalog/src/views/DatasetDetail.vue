<template>
  <div>
    <router-link :to="{ name: 'do-catalog' }">Datasets list</router-link>
    <DatasetHeader></DatasetHeader>
    <div class="grid grid-cell u-flex__justify--center">
      <NavigationTabs class="grid-cell--col10">
        <router-link :to="{ name: 'do-dataset-summary' }">Summary</router-link>
        <router-link :to="{ name: 'do-dataset-data' }">Data</router-link>
      </NavigationTabs>
    </div>
    <router-view :key="$route.fullPath"></router-view>
  </div>
</template>

<script>
import DatasetHeader from '../components/datasetDetail/DatasetHeader';
import NavigationTabs from '../components/datasetDetail/NavigationTabs';

export default {
  name: 'DatasetDetail',
  mounted() {
    this.fetchDataset();
    this.fetchKeyVariables();
  },
  components: {
    DatasetHeader,
    NavigationTabs
  },
  methods: {
    fetchDataset() {
      this.$store.dispatch(
        'doCatalog/fetchDataset',
        this.$route.params.datasetId
      );
    },
    fetchKeyVariables() {
      this.$store.dispatch(
        'doCatalog/fetchKeyVariables',
        this.$route.params.datasetId
      );
    }
  }
};
</script>
