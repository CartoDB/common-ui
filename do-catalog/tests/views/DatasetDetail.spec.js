import { createLocalVue, shallowMount } from "@vue/test-utils";
import Vuex from 'vuex';
import DatasetDetail from "@/views/DatasetDetail";
import VueRouter from 'vue-router';

const localVue = createLocalVue();
localVue.use(VueRouter);
localVue.use(Vuex);
const router = new VueRouter();

describe("DatasetDetail.vue", () => {
  let actions;
  let store;

  beforeEach(() => {
    actions = {
      "catalog/fetchDataset": jest.fn()
    };
    store = new Vuex.Store({
      state: {},
      actions
    });
  });

  it("Renders view properly", () => {
    const datasetDetail = shallowMount(DatasetDetail, {
      localVue,
      router,
      store
    });
    expect(datasetDetail).toMatchSnapshot();
  });

  it("Fetch data from the api", () => {
    const datasetDetail = shallowMount(DatasetDetail, {
      localVue,
      router,
      store
    });
    expect(actions["catalog/fetchDataset"]).toHaveBeenCalled();
  });
});
