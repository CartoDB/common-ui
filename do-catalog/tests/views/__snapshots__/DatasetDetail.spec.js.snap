// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatasetDetail.vue Renders view properly 1`] = `
<div id="app">
  <datasetheader-stub></datasetheader-stub>
  <div class="grid grid-cell u-flex__justify--center">
    <navigationtabs-stub class="grid-cell--col10">
      <router-link-stub to="[object Object]" tag="a" event="click">Summary</router-link-stub>
      <router-link-stub to="[object Object]" tag="a" event="click">Data</router-link-stub>
      <router-link-stub to="/" tag="a" event="click">Map</router-link-stub>
      <router-link-stub to="/" tag="a" event="click">Use it</router-link-stub>
      <router-link-stub to="/" tag="a" event="click">Catalog</router-link-stub>
    </navigationtabs-stub>
  </div>
  <router-view-stub name="default"></router-view-stub>
</div>
`;
